gb-cpu/src/address_bus/operation.rs

```diff
@@ -0,0 +1,95 @@
+use super::{Error, Position};
+use rand::{rngs::SmallRng, Rng, SeedableRng};
+use std::cell::RefCell;
+
+/// RomOperation basic trait to implement for a ROM Emulator.
+/// Rom is generally Read-only so `write` is not often used
+pub trait RomOperation {
```

Comment:
Is there a reason to implement two different traits with the exact same properties? A trait with read and write implemented differently for Rom and File would be better i think

Reply:
Yes, there's a reason and the reason is `Memory Bank Controller` that handle both the rom and the ram bank.